plugins/romloader/uart/romloader_uart_main.cpp

```diff
@@ -206,6 +207,98 @@ romloader_uart::~romloader_uart(void)
 	}
 }
 
+/*
+* This function is used to check if the netX is stuck in an endless loop waiting for an acknowledge for a packet and re-sending it
+* If it is the case we send a cancel_operation packet that should be able to to get the netX out of the loop
+*/
+bool romloader_uart::fix_deadloop()
+{
+	size_t sizTransfered;
+	bool fResult = false;
+	uint8_t aucData[32];
+	MIV3_PACKET_HEADER_T *ptPacketHeader;
+	uint8_t ucRetries = 5;
+	uint32_t ulDataReceived;
+	
+	
+	do
+	{
+		
+		// clear buffer until we timout after 200ms (meaning we are between two received packets) or we reach a max transferred data size
+		ulDataReceived = 0;
+		do
+		{
+			sizTransfered = m_ptUartDev->RecvRaw(aucData, 1, 200);
+			ulDataReceived += sizTransfered;
+		} while( sizTransfered==1 || ulDataReceived > 100);
```

Comment:
Maybe the receive length should be larger:
#define UART_MAXIMUM_PACKET_SIZE 2048